x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/ElasticsearchInternalService.java

```diff
@@ -127,22 +129,38 @@ private void customElandCase(
                         + "]. You may need to load it into the cluster using eland."
                 );
             } else {
-                var customElandInternalServiceSettings = (CustomElandInternalServiceSettings) CustomElandInternalServiceSettings.fromMap(
-                    serviceSettingsMap
-                ).build();
-                throwIfNotEmptyMap(serviceSettingsMap, name());
+                var model = createCustomElandModel(
+                    inferenceEntityId,
+                    taskType,
+                    serviceSettingsMap,
+                    taskSettingsMap,
+                    ConfigurationParseContext.REQUEST
+                );
 
-                var taskSettings = CustomElandModel.taskSettingsFromMap(taskType, taskSettingsMap);
+                throwIfNotEmptyMap(serviceSettingsMap, name());
                 throwIfNotEmptyMap(taskSettingsMap, name());
 
-                var model = CustomElandModel.build(inferenceEntityId, taskType, name(), customElandInternalServiceSettings, taskSettings);
                 delegate.onResponse(model);
             }
         });
 
         client.execute(GetTrainedModelsAction.INSTANCE, request, getModelsListener);
     }
 
+    private static CustomElandModel createCustomElandModel(
+        String inferenceEntityId,
+        TaskType taskType,
+        Map<String, Object> serviceSettings,
+        Map<String, Object> taskSettings,
+        ConfigurationParseContext context
+    ) {
+        return switch (taskType) {
+            case TEXT_EMBEDDING -> new CustomElandEmbeddingModel(inferenceEntityId, taskType, NAME, serviceSettings, context);
+            case RERANK -> new CustomElandRerankModel(inferenceEntityId, taskType, NAME, serviceSettings, taskSettings, context);
+            default -> throw new ElasticsearchStatusException(TaskType.unsupportedTaskTypeErrorMsg(taskType, NAME), RestStatus.BAD_REQUEST);
```

Comment:
I believe these are the only task types we support because the `infer()` call will throw an exception if it isn't one of these values.